### PR TITLE
Log stacktrace to stderr

### PR DESCRIPTION
**Pull request by nkottary**
Replace `println` with `@error`

### CODE REVIEW

File: src/register.jl [hunk 1]
```diff
@@ -703,8 +703,9 @@ function register(
 
         clean_registry = false
     catch ex
-        println(get_backtrace(ex))
-        add!(status, :unexpected_registration_error)
+        @error("Unexpected error while registering", stacktrace=get_backtrace(ex))
+        err = :unexpected_registration_error
```
Comment (GunnarFarneback) [inline src/register.jl:707]: `@error` looks very reasonable but I'm not sure that naming the `err` argument improves anything. Mostly makes the code longer and makes you wonder why it was named and if it's maybe coming from an outer scope and used later.

Comment (nkottary) [reply src/register.jl:707]: Its used in the `finally` block below.

Comment (GunnarFarneback) [reply src/register.jl:707]: That's funny. It does come from an outer scope and it is used later. How long has this been broken? (No, I'm not sure I dare look at the git blame.)

Comment (nkottary) [reply src/register.jl:707]: What is broken exactly?

Comment (ericphanson) [reply src/register.jl:707]: until this PR, `err` is left as `nothing`, so it was broken but you're fixing it (I think)

Comment (GunnarFarneback) [reply src/register.jl:707]: Exactly my thinking and it was probably I who left `err` hanging. When I look back at old code it once played a much larger role. Actually at this point I don't see why `finally` is not just merged with `catch` and the `clean_registry` and `err` stuff can be simplified. But that has nothing to do with this PR.

**Pull request merged**